ste/xfer-anyToRemote-file.go

```diff
@@ -165,15 +165,8 @@ func anyToRemote(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, sen
 			// if src does not have snapshot/versionId, then error out as we cannot copy into the object itself
 			// if dst has snapshot or versionId specified, do not error and let the service fail the request with clear message
 			srcRQ := srcURL.Query()
-			dstRQ := dstURL.Query()
 
-			// note that query is now all lower case at this point, extract strings from query
-			srcShareSnapshot := strings.Join(srcRQ["sharesnapshot"], "")
-			dstShareSnapshot := strings.Join(dstRQ["sharesnapshot"], "")
-			srcVID := strings.Join(srcRQ["versionid"], "")
-			dstVID := strings.Join(dstRQ["versionid"], "")
-
-			if srcShareSnapshot == dstShareSnapshot && srcVID == dstVID {
+			if !(len(srcRQ["sharesnapshot"]) > 0 || len(srcRQ["snapshot"]) > 0 || len(srcRQ["versionid"]) > 0) {
```

Comment:
Minor style issue: for readability it might be best to avoid the leading ! and instead do:

len(srcRQ["sharesnapshot"]) == 0 && len(srcRQ["snapshot"])  == 0 && len(srcRQ["versionid"])  == 0